RecoTauTag/RecoTau/src/PFRecoTauClusterVariables.cc

```diff
@@ -0,0 +1,367 @@
+          sum_pt += pt;        
+          if (var == 0)
+            sum_dx_pt += pt * dr;
+          else if (var == 1)
+            sum_dx_pt += pt * deta;
+          else if (var == 2)
+            sum_dx_pt += pt * dphi;
+        }
+      }
+    }	 
+    if (sum_pt > 0.){
+      return sum_dx_pt/sum_pt;
+    }  
+    return 0.;	
+  }
+  /// return sum of pt weighted values of dr relative to tau candidate for all pf photon candidates,
+  /// which are associated to signal
+  float tau_pt_weighted_dr_signal(const reco::PFTau& tau, int dm) {
+    return pt_weighted_dx(tau, 0, 0, dm);
+  }
+  float tau_pt_weighted_dr_signal(const pat::Tau& tau, int dm) {
+    return pt_weighted_dx(tau, 0, 0, dm);
+  }
+  /// return sum of pt weighted values of deta relative to tau candidate for all pf photon candidates,
+  /// which are associated to signal
+  float tau_pt_weighted_deta_strip(const reco::PFTau& tau, int dm) {
+    return pt_weighted_dx(tau, dm==10 ? 2 : 1, 1, dm);
+  }
+  float tau_pt_weighted_deta_strip(const pat::Tau& tau, int dm) {
+    return pt_weighted_dx(tau, dm==10 ? 2 : 1, 1, dm);
+  }
+  /// return sum of pt weighted values of dphi relative to tau candidate for all pf photon candidates,
+  /// which are associated to signal
+  float tau_pt_weighted_dphi_strip(const reco::PFTau& tau, int dm) {
+    return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
+  }
+  float tau_pt_weighted_dphi_strip(const pat::Tau& tau, int dm) {
+    return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
+  }  
+  /// return sum of pt weighted values of dr relative to tau candidate for all pf photon candidates,
+  /// which are inside an isolation conde but not associated to signal
+  float tau_pt_weighted_dr_iso(const reco::PFTau& tau, int dm) {
+    return pt_weighted_dx(tau, 2, 0, dm);
+  }
+  float tau_pt_weighted_dr_iso(const pat::Tau& tau, int dm) {
+    return pt_weighted_dx(tau, 2, 0, dm);
+  }
+  /// return total number of pf photon candidates with pT>500 MeV, which are associated to signal
+  unsigned int tau_n_photons_total(const reco::PFTau& tau) {
+    unsigned int n_photons = 0;
+    for (auto& cand : tau.signalPFGammaCands()) {
+      if ((float)cand->pt() > 0.5)
+        ++n_photons;
+    }
+    for (auto& cand : tau.isolationPFGammaCands()) {
+      if ((float)cand->pt() > 0.5)
+        ++n_photons;
+    }
+    return n_photons;
+  }
+  unsigned int tau_n_photons_total(const pat::Tau& tau) {
+    unsigned int n_photons = 0;
+    for (auto& cand : tau.signalGammaCands()) {
+      if (cand->pt() > 0.5) 
+        ++n_photons;
+    }
+    for (auto& cand : tau.isolationGammaCands()) {  
+      if (cand->pt() > 0.5)
+	++n_photons;
+    }
+    return n_photons;
+  }  
+
+  bool fillMVAInputs(float* mvaInput, const pat::Tau& tau, int mvaOpt, const std::string nameCharged, const std::string   nameNeutral, const std::string namePu, const std::string nameOutside, const std::string nameFootprint) 
```

Comment:
should I take this replacement of the implementation of PATTauDiscriminationByMVAIsolationRun2 as a move to the new standard single supported MVA? 
If the other variants are expected to remain supported, then a simple ```fillMVAInputs``` method name here is not enough. In this case please make a distinction with some specification in the function name.

Reply:
We are a bit confused by the comment: PATTauDiscriminationByMVAIsolationRun2 plugin is the single implementation of MVA-based isolation tau-Ids with the `fillMVAInputs` method having itself all the different MVAIso variants included. Could you please clarify what you exactly mean?

Regardless, we can rename the method to `fillIsoMVARun2Inputs` (`Iso` and `Run2` added) to better distinguish from other possible MVA-based tau-ids, e.g. anti electron ones, which could in principle use the cluster variables.

Reply:
I think I'm still confused - the method as implemented works for all tau MVAs by means of the "mvaOpt" argument. 

Concerning dropping the possibility to recalculate previous versions of the MVA from the tau code, I'm in principle in favour of this since it will make the code simpler, but it's not the focus of this PR.

Reply:
OK, thank you for clarification. 

Just for notice, there are the following MVA-based tau-Ids and related plugins:
* MVA-based isolation:
   * `PFRecoTauDiscriminationByMVAIsolation2.cc` for Run-1
   * `PFRecoTauDiscriminationByMVAIsolationRun2.cc` for Run-2 with its counterpart for `pat::Taus` called `PATTauDiscriminationByMVAIsolationRun2.cc`
* e->tauh fakes veto:
   * `PFRecoTauDiscriminationAgainstElectronMVA5.cc` for Run-1
   * `PFRecoTauDiscriminationAgainstElectronMVA6.cc` for Run-2 with its counterpart for `pat::Taus` called `PATTauDiscriminationAgainstElectronMVA6.cc`
* mu->tauh fakes veto:
   * `PFRecoTauDiscriminationAgainstMuonMVA.cc`

The old Run-1 plugins can be removed at some point, but it is beyond a scope of this PR. 

To avoid confusion we will also rename the method to `fillIsoMVARun2Inputs` to indicate that it is used for this particular MVA-based tau-Id.